### PR TITLE
348 individual osc input messages

### PR DESCRIPTION
**Pull request by jvlyndark**
This PR contains the individual OSC message changes as well as most of the changes made for the refactoring of the OSC to unify with the Q_PROPERTY system. Final edits to both tickets will be pushed in the subsequent PR.

### CODE REVIEW

File: autonomx/ComputeEngine.cpp [hunk 1]
```diff
@@ -85,233 +121,119 @@ void ComputeEngine::receiveOscData(int generatorId, QVariantList data) {
...
+    if(parameterControlList.contains(parameter1)) {
+        qDebug() << "executes in catch-all";
+        controlMessageArray = parameter1.toLocal8Bit();
+        controlMessageArrayChar = controlMessageArray.data();
+        generator->setProperty(controlMessageArrayChar, inputValue);
+        return;
     }
+
+    ///////// Only Reaches here if message did not fit any acceptable OSC Input Message format ///////////
+    qDebug() << "ERROR: Invalid OSC Input Message!";
```
Comment (aalex) [inline autonomx/ComputeEngine.cpp]: We should debug-print the contents of the invalid messages we receive.

File: autonomx/Generator.cpp [hunk 2]
```diff
@@ -680,26 +686,28 @@ void Generator::applyInputRegion() {
...
-            for(int x = region->getRect().x(); x < xMax; x++) {
+        for(int x = region->getRect().x(); x < xMax; x++) {
             for(int y = region->getRect().y(); y < yMax; y++) {
-                QString wolframCA = "WolframCA";
-                QString gameOfLife = "GameOfLife";
                 double valueToWrite;
                 double inputValue;
-                if(this->getType() == wolframCA || this->getType() == gameOfLife) {
+                // rounds values to integers if generator type is WolframCA or GameOfLife
+                if(this->getType() == "WolframCA" || this->getType() == "GameOfLife") {
```
Comment (aalex) [inline autonomx/Generator.cpp:694]: Use constants for the type names, such as "GameOfList" or "WolframCA". (static const char*)

**Review by aalex — DISMISSED** [review]
- We should debug-print the contents of the invalid messages we receive.
- Use constants for the type names, such as "GameOfLife" or "WolframCA". (static const char*) https://stackoverflow.com/questions/1639154/how-to-declare-a-static-const-char-in-your-header-file
- Maybe squash and merge once your PR is ready to be merged?

I am approving your PR, but I think the changes above should be done as soon as possible, and ideally before merging it.

File: autonomx/Generator.cpp (hunk 2 above)
Comment (jvlyndark) [reply autonomx/Generator.cpp:694]: Hey, sorry, I'm not sure I understand how to do what you're suggesting. Are you saying I shouldn't use the "this->getType()" or the QString? The linked thread on stackoverflow was confusing to me in relation to this.

**Comment by jvlyndark** [comment]
@aalex changes made (except for the static const char*, which I will do in the next PR)

**Review by aalex — APPROVED** [review]

**Pull request merged**